concurrent/src/main/java/com/facebook/airlift/concurrent/ConcurrentScheduledExecutor.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.concurrent;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ConcurrentScheduledExecutor
+        implements ScheduledExecutorService
+{
+    private final List<ScheduledExecutorService> schedulers;
+
+    public ConcurrentScheduledExecutor(
+            int schedulerCount,
+            int threadsPerScheduler,
+            String threadBaseName)
+    {
+        this(schedulerCount, threadsPerScheduler, threadBaseName, false);
+    }
+
+    public ConcurrentScheduledExecutor(
+            int schedulerCount,
+            int threadsPerScheduler,
+            String threadBaseName,
+            boolean removeOnCancel)
+    {
+        checkArgument(schedulerCount > 0, "schedulerCount must be at least one");
+        checkArgument(threadsPerScheduler > 0, "threadsPerScheduler must be at least one");
+        requireNonNull(threadBaseName, "threadBaseName is null");
+
+        ImmutableList.Builder<ScheduledExecutorService> schedulersBuilder = ImmutableList.builder();
+        for (int i = 0; i < schedulerCount; i++) {
+            ThreadFactory threadFactory = daemonThreadsNamed(threadBaseName + format("-scheduler-%d", i) + "-%s");
+            ScheduledThreadPoolExecutor scheduledExecutorService = new ScheduledThreadPoolExecutor(threadsPerScheduler, threadFactory);
+            if (removeOnCancel) {
```

Comment:
nit: It can be simplified to `scheduledExecutorService.setRemoveOnCancelPolicy(removeOnCancel)`